spanner/tests/unit/test_snapshot.py

```diff
@@ -31,6 +33,68 @@
 PARAMS_WITH_BYTES = {'bytes': b'DEADBEEF'}
 
 
+class Test_restart_on_unavailable(unittest.TestCase):
+
+    def _call_fut(self, restart):
+        from google.cloud.spanner.snapshot import _restart_on_unavailable
+
+        return _restart_on_unavailable(restart)
+
+    def _make_item(self, value, resume_token=''):
+        return mock.Mock(
+            value=value, resume_token=resume_token,
+            spec=['value', 'resume_token'])
+
+    def test_iteration_w_empty_raw(self):
+        ITEMS = ()
+        raw = _MockIterator()
+        restart = mock.Mock(spec=[], return_value=raw)
+        resumable = self._call_fut(restart)
+        self.assertEqual(list(resumable), [])
+
+    def test_iteration_w_non_empty_raw(self):
+        ITEMS = (self._make_item(0), self._make_item(1))
+        raw = _MockIterator(*ITEMS)
+        restart = mock.Mock(spec=[], return_value=raw)
+        resumable = self._call_fut(restart)
+        self.assertEqual(list(resumable), list(ITEMS))
+        restart.assert_called_once_with()
+
+    def test_iteration_w_raw_w_resume_tken(self):
+        ITEMS = (
+            self._make_item(0),
+            self._make_item(1, resume_token='DEADBEEF'),
+            self._make_item(2),
+            self._make_item(3),
+        )
+        raw = _MockIterator(*ITEMS)
+        restart = mock.Mock(spec=[], return_value=raw)
+        resumable = self._call_fut(restart)
+        self.assertEqual(list(resumable), list(ITEMS))
+        restart.assert_called_once_with()
+
+    def test_iteration_w_raw_raising_unavailable(self):
```

Comment:
can  you also add a test case where it raises unavailable just after getting a resume token.

Reply:
fa7bf9e